evm_loader/emulator/src/account_storage.rs

```diff
@@ -0,0 +1,335 @@
+                },
+                Err(_) => {
+                    eprintln!("Account not found {}", &address.to_string());
+
+                    new_accounts.insert(address);
+
+                    false
+                }
+            }
+        } else {
+            true
+        }
+    }
+
+    // pub fn make_solidity_account<'a>(self, account:&'a SolanaAccount) -> SolidityAccount<'a> {
+    //     let mut data = account.account.data.clone();
+    //     let data_rc: std::rc::Rc<std::cell::RefCell<&mut [u8]>> = Rc::new(RefCell::new(&mut data));
+    //     SolidityAccount::new(&account.key, data_rc, account.account.lamports).unwrap()
+    // }
+
+    pub fn apply<A, I>(&self, values: A)
+            where
+                A: IntoIterator<Item=Apply<I>>,
+                I: IntoIterator<Item=(H256, H256)>,
+    {             
+        let mut accounts = self.accounts.borrow_mut(); 
+
+        for apply in values {
+            match apply {
+                Apply::Modify {address, basic, code: _, storage: _, reset_storage} => {
+                    match accounts.get_mut(&address) {
+                        Some(acc) => {
+                            *acc.writable.borrow_mut() = true;
+                        },
+                        None => {
+                            eprintln!("Account not found {}", &address.to_string());
+                        },
+                    }
+                    eprintln!("Modify: {} {} {} {}", &address.to_string(), &basic.nonce.as_u64(), &basic.balance.as_u64(), &reset_storage.to_string());
+                },
+                Apply::Delete {address: addr} => {
+                    eprintln!("Delete: {}", addr.to_string());
+                },
+            }
+        };
+    }
+
+    pub fn get_used_accounts(&self, status: &String, result: &std::vec::Vec<u8>)
+    {          
+        let mut arr = Vec::new();    
+        
+        eprint!("[");
+        let accounts = self.accounts.borrow();
+        for (address, acc) in accounts.iter() {
+            arr.push(AccountJSON{address: "0x".to_string() + &hex::encode(&address.to_fixed_bytes()), writable: acc.writable, new: false});
+            eprint!("{{\"address\":\"0x{}\",\"write\":\"{}\"}},", &hex::encode(&address.to_fixed_bytes()), &acc.writable.to_string());
+        }
+        let new_accounts = self.new_accounts.borrow(); 
+        for address in new_accounts.iter() {
+            arr.push(AccountJSON{address: "0x".to_string() + &hex::encode(&address.to_fixed_bytes()), writable: false, new: true});
+            eprint!("{{\"address\":\"0x{}\",\"new\":\"true\"}},", &hex::encode(&address.to_fixed_bytes()));
+        }    
+        eprintln!("]");
+
+        let js = json!({"accounts": arr, "result": &hex::encode(&result), "exit_status": &status}).to_string();
+
+        println!("{}", js);
+    }
+}
+
+impl AccountStorage for EmulatorAccountStorage {
+    fn contract_id(&self) -> H160 { self.contract_id }
+    fn block_number(&self) -> U256 {
```

Comment:
`block_number` & `block_timestamp` need to get in EmulatorAccountStorage initialization and return cached values.